test/torch/tensors/test_promise.py

```diff
@@ -0,0 +1,157 @@
+import pytest
+import torch
+import syft
+
+
+def test__str__():
+    a = syft.Promises.FloatTensor(shape=torch.Size((3, 3)))
+    assert isinstance(a.__str__(), str)
+
+
+@pytest.mark.parametrize("cmd", ["__add__", "sub", "__mul__"])
+def test_operations_between_promises(hook, cmd):
+    hook.local_worker.is_client_worker = False
+
+    a = syft.Promises.FloatTensor(shape=torch.Size((2, 2)))
+    b = syft.Promises.FloatTensor(shape=torch.Size((2, 2)))
+
+    actual = getattr(a, cmd)(b)
+
+    ta = torch.tensor([[1.0, 2], [3, 4]])
+    tb = torch.tensor([[-8.0, -7], [6, 5]])
+    a.keep(ta)
+    b.keep(tb)
+
+    expected = getattr(ta, cmd)(tb)
+
+    assert (actual.value() == expected).all()
+
+    hook.local_worker.is_client_worker = True
+
+
+@pytest.mark.parametrize("cmd", ["__add__", "sub", "__mul__"])
+def test_operations_with_concrete(hook, cmd):
+    hook.local_worker.is_client_worker = False
+
+    a = syft.Promises.FloatTensor(shape=torch.Size((2, 2)))
+    b = torch.tensor([[-8.0, -7], [6, 5]]).wrap()  # TODO fix need to wrap
+
+    actual = getattr(a, cmd)(b)
+
+    ta = torch.tensor([[1.0, 2], [3, 4]]).wrap()
+    a.keep(ta)
+
+    expected = getattr(ta, cmd)(b)
+
+    assert (actual.value() == expected).all()
+
+    hook.local_worker.is_client_worker = True
+
+
+def test_send(workers):
+    bob = workers["bob"]
+
+    a = syft.Promises.FloatTensor(shape=torch.Size((2, 2)))
+
+    x = a.send(bob)
+
+    x.keep(torch.ones((2, 2)))
+
+    assert (x.value().get() == torch.ones((2, 2))).all()
+
+
+@pytest.mark.parametrize("cmd", ["__add__", "sub", "__mul__"])
+def test_remote_operations(workers, cmd):
```

Comment:
Can we also check that .move() works with Promises as well, meaning you could chain promises across a long series of workers and they would work?

Reply:
Hum, that's right, I didn't think about .move(), it does not work for now.
Can you give a small example of what you mean by chaining promises across a series of workers?

Reply:
Sure thing!

The first approach is via .send() which sends the pointers around.
```
x = sy.Promises.FloatTensor([2,2])
x_bob = x.send(bob)
x_john = x_bob.send(john)
x_bill = x_john.send(bill)
```
So now we have a promise pointing to bill, which is a promise pointing to john, which is a promise pointing to bob, and if we keep the first one it'll send the data all the way down the chain.
```x_bill.keep(th.zeros([2,2])```

We can do something identical using .move()
```
x = sy.Promises.FloatTensor([2,2])
x_bob = x.send(bill)
x_john = x_bob.move(john)
x_bill = x_john.move(bill)
x_bob = x_bill.move(bob)
```
This is identical, but as you can see we can construct it in reverse. Often the order matters. I hope this is useful.

Reply:
status on this?

Reply:
Currently, when you use `.send()` on a Promise, you don't get a promise but a pointer pointing to a promise.
Was it a typo on your side or do we have a different understanding on what should happen when sending promises?
And currently, using `.move()`moves the promise to another worker but shouldn't create a chain as with consecutive sends, am I wrong?

Reply:
Ahh, I see what you mean by consecutive sends. I think that this can work. I'm happy for now but we might modify this in the future.